test/api/valik/search/search_setup_test.cpp

```diff
@@ -70,3 +72,216 @@ TEST(pattern_begin_positions, extra_overlap)
 
     EXPECT_EQ(begin_positions, expected); // seen all positions
 }
+
+// ====================================================================================
+// seq = CGCAAAACGCGGC      minimisers = [AAAA, AAAC, AACG]
+// 	w = 8
+// 	k = 4
+//
+// minimiser	             span_begin    minimiser belongs to windows starting at
+// AAAA		CGCAAAACGCG          0                      0, 1, 2, 3
+// AAAC		    AAACGCGG         4                          4
+// AACG		     AACGCGGC        5                          5
+//
+// ====================================================================================
+
+TEST(make_pattern_bounds, first_pattern_of_query)
+{
+    // normal case where pattern starts from beginning of query
+    valik::search_arguments arguments{};
+    arguments.pattern_size = 12;
+    arguments.window_size = 8;
+    arguments.kmer_size = 4;
+    arguments.errors = 1;
+
+    std::vector<size_t> const & window_span_begin{0, 4, 5};
+
+    size_t const pattern_begin = 0;
+    valik::pattern_bounds expected{};
+
+
+    // |CGCAAAACGCGG|C
+    //
+    // 0: |[CGCAAAAC]GCGG|C     consider
+    // 1: |C[GCAAAACG]CGG|C     consider
+    // 2: |CG[CAAAACGC]GG|C     consider
+    // 3: |CGC[AAAACGCG]G|C     consider
+    // 4: |CGCA[AAACGCGG]|C     consider
+    // 5: |CGCAA[AACGCGG |C]     don't consider
```

Comment:
Which of them are the minimiser? This is important since `begin_position` and `end_position` are relative to the minimiser positions.